store/generated_key/v2/eigenda.go

```diff
@@ -64,12 +64,18 @@ func (e Store) Get(ctx context.Context, key []byte) ([]byte, error) {
 		return nil, fmt.Errorf("RLP decoding EigenDA v2 cert: %w", err)
 	}
 
-	payload, err := e.retriever.GetPayload(ctx, &cert)
-	if err != nil {
-		return nil, fmt.Errorf("getting payload: %w", err)
+	// Try each retriever in sequence until one succeeds
+	var errs []error
+	for _, retriever := range e.retrievers {
```

Comment:
There's a lot of code in this PR to possibly make the retrieval method more flexible in the future (list instead of just hardcoding 2 methods, flag that is unused just to set the config, etc.)

Do we really need this flexibility? Should we consider only adding it in the future if/when we need it? Is the idea that we would eventually want to add more retrieval methods, such as unifying all the different backends we currently have (s3, redis) as more retrievers in the future?

Reply:
> Is the idea that we would eventually want to add more retrieval methods, such as unifying all the different backends we currently have (s3, redis) as more retrievers in the future?

Yes, exactly. Pursuing this strategy is at the top of my proxy wishlist, as I think would *greatly* simplify things.

I think it makes sense to introduce this flexibility now, rather than adding additional hard coded stuff that would make this step more difficult in the future. I also don't think this flexibility cost us very much in terms of code complexity: this PR would be only marginally simpler if we went the hardcoded route